scipy/spatial/_polygon_surface_area.py

```diff
@@ -0,0 +1,79 @@
+"""
+
+Polygon Surface Area Code
+
+.. versionadded:: 1.0.0
+
+"""
+
+import numpy as np
+from scipy.spatial.distance import pdist
+from . import _surface_area
+
+#
+# Copyright (C)  James Nichols and Tyler Reddy
+# Contributions from Tyler Reddy are owned by LANL
+# and respect the scipy distribution license
+#
+# Distributed under the same BSD license as Scipy.
+
+def _vertex_index_strider(index, num_vertices):
+    # handle the wrapping / iteration over
+    # polygon vertices in either CW or CCW
+    # sort order
+    forward_index = index + 1
+    backward_index = index - 1
+    if forward_index > (num_vertices - 1):
+        forward_index = 0
+    return forward_index, backward_index
+
+def poly_area(vertices, radius=None, threshold=1e-21,
+              cython=None):
+    # calculate the surface area of a planar or spherical polygon
+    # crude pure Python implementation for handling a single
+    # polygon at a time
+    # based on JPL Publication 07-3 by Chamberlain and Duquette (2007)
+    # for planar polygons we currently still require x,y,z coords
+    # can just set i.e., z = 0 for all vertices
+    if pdist(vertices).min() < threshold:
+        raise ValueError("Duplicate vertices detected within threshold.")
+
+    num_vertices = vertices.shape[0]
+    area_sum = 0
+
+    if radius is not None: # spherical polygons
```

Comment:
I might refactor this to split out two pure-python helper methods: `_spherical_polygon_area` and `_planar_polygon_area`. Then this function would just dispatch to the correct helper based on `cython` and `radius is None` condtions. This would make testing the cython versions easier, too.